### PR TITLE
Updated trip CRUD UX in Nova

### LINKED ISSUE #2674: Show what documents are included in DocuSign packet for travel
This is non-obvious and I have seen people ask for forms over Slack that are already submitted via DocuSign.

### LINKED ISSUE #3491: Rename "TAR Recieved" metric to "Forms Received"

### LINKED ISSUE #3557: Search box for primary contact on travel should be filtered
Also maybe PMs on teams

### LINKED ISSUE #4019: Clarify that travel fields are per-person not total
Can we also add some validation to the trip fee field to check the math?

### PR DESCRIPTION
**Pull request by kberzinch**
- Nova trip CRUD UI mostly replaced (closes #4019)
- DocuSign toggles are now in a dedicated `forms` column on the trip instead of reusing `tar_required` and `tar_transportation_mode` (closes #2674)
    - Other code that relied on those columns has been updated appropriately
- "TAR Received" metric renamed to "Forms Received" (closes #3491)
- Project manager search on teams and primary contact search on trips is now filtered to `team-lead`s, `project-manager`s, and `officer`s only (closes #3557)

Partial progress towards #3488, #2618

### CODE REVIEW

File: app/Nova/Travel.php [hunk 1]
```diff
@@ -78,315 +80,387 @@ class Travel extends Resource
...
+                    ->rules('sometimes')
+                    ->help(
+                        'Ask the treasurer for the correct value for this field.'
+                    )
+                    ->hideFromIndex(),
+
+                Select::make('Department', 'department_number')
+                    ->options(DepartmentNumbers::DESCRIPTIONS)
+                    ->displayUsingLabels()
+                    ->searchable()
```
Comment (kberzinch) [inline app/Nova/Travel.php:402]: This is I think just using basic substring search on the frontend. We can make this fuzzy search with a model like we did for airports but that feels like overkill.

File: app/Nova/User.php [hunk 2]
```diff
@@ -809,4 +810,25 @@ private static function adminCanRun(NovaRequest $request): bool
...
     }
+
+    /**
+     * Build a Scout search query for the given resource.
+     *
+     * @param  \Laravel\Scout\Builder  $query
+     */
+    public static function scoutQuery(NovaRequest $request, $query): \Laravel\Scout\Builder
+    {
+        if ($request->field === 'projectManager' || $request->field === 'primaryContact') {
```
Comment (kberzinch) [inline app/Nova/User.php:821]: This is going to be a little weird in test because the Meilisearch index is out of date. Lmk if I need to import things.

**Review by JoshOldenburg — CHANGES_REQUESTED** [review]
A few minor things but overall looks good. Not my most thorough review ever but at this point something is better than nothing.

File: app/Nova/Travel.php [hunk 3]
```diff
@@ -78,315 +80,387 @@ class Travel extends Resource
...
+                ->sortable()
+                ->required()
+                ->rules('required')
+                ->searchable()
+                ->withoutTrashed(),
 
             Date::make('Departure Date')
+                ->sortable()
                 ->required()
                 ->rules('required', 'date', 'before:return_date'),
```
[the remark this answers is not in the document]
Comment (kberzinch) [reply app/Nova/Travel.php:164]: I'm inclined to leave it as-is for now, it works for all of the trips that are currently logged.

File: app/Nova/Travel.php [hunk 4]
```diff
@@ -449,12 +523,50 @@ public function cards(Request $request): array
...
+        }
+
+        // validate account and department make sense
+        if ($request->tar_project_number !== null && $request->department_number !== null) {
+            if ($request->tar_project_number === 'GTF250000211' && $request->department_number !== '250') {
+                $validator->errors()->add(
+                    'department_number',
+                    'The selected department and Workday account numbers do not match.'
+                );
+            }
```
[the remark this answers is not in the document]
Comment (kberzinch) [reply app/Nova/Travel.php:642]: GTF is the legal entity but the accounts are held on behalf of specific departments. GTF250000211 is held on behalf of Mechanical Engineering, hence the 250 in the account number.

Again, this works for most trips, and if it becomes a problem we can revisit it.

File: app/Nova/Travel.php [hunk 5]
```diff
@@ -78,318 +84,417 @@ class Travel extends Resource
...
+     * @phan-suppress PhanTypeInvalidCallableArraySize
      */
     public function fields(Request $request): array
     {
         return [
-            Text::make('Event Name', 'name')
+            Text::make('Trip Name', 'name')
+                ->help(view('nova.help.travel.tripname')->render())
                 ->sortable()
-                ->help('This should typically be the name of the competition followed by the year.')
```
[the remark this answers is not in the document]
Comment (kberzinch) [reply app/Nova/Travel.php:89]: I think I was fiddling around with adding more guidance here but didn't end up doing that.

File: app/Nova/Travel.php [hunk 6]
```diff
@@ -78,315 +80,387 @@ class Travel extends Resource
...
+                                    ->rules('max:30');
+                            }
+                        }
+                    )
+                    ->rules('sometimes')
+                    ->hide()
+                    ->showOnDetail(
+                        fn (): bool => $this->showFieldOnDetail(\App\Models\Travel::TRAVEL_INFORMATION_FORM_KEY)
+                    )
+                    ->help('If you are not staying overnight, leave this field blank.')
```
[the remark this answers is not in the document]
Comment (kberzinch) [reply app/Nova/Travel.php:279]: What other fields are per se optional?

[the remark this answers is not in the document]
Comment (JoshOldenburg) [reply app/Nova/Travel.php:279]: Never mind, I was looking at the costs but they say to enter 0 instead of leaving it blank.

**Review by JoshOldenburg — APPROVED** [review]

**Review by evan10s — APPROVED** [review]

**Pull request merged**